corehq/apps/users/util.py

```diff
@@ -387,3 +392,58 @@ def _get_changed_details(couch_user, action, fields_changed):
     for prop in USER_FIELDS_TO_IGNORE_FOR_HISTORY:
         changed_details.pop(prop, None)
     return changed_details
+
+
+def bulk_auto_deactivate_commcare_users(user_ids, domain):
+    """
+    Deactivates CommCareUsers in bulk. Please pre-chunk ids to a reasonable size.
+    Also please reference the save() method in CommCareUser when making changes.
+    :param user_ids: list of user IDs
+    :param domain: name of domain user IDs belong to
+    """
+    from corehq.apps.users.models import UserHistory, CommCareUser
+    from corehq.apps.users.model_log import UserModelAction
+
+    user_docs_to_bulk_save = []
+    for user_doc in get_docs(CommCareUser.get_db(), keys=user_ids):
+        if user_doc['is_active']:
+            user_doc['is_active'] = False
+            user_doc['last_modified'] = json_format_datetime(datetime.datetime.utcnow())
```

Comment:
What do you think of using a single timestamp for the entire batch (put this date in a variable initialized outside of the loop)? I think it might be nice for auditing purposes.